### PR TITLE
Add support for run image mirrors

### PR DESCRIPTION
**Pull request by thisisnotashwin**
- Add run image mirrors into builder spec to accept additional mirrors
- Select best run image from list of metadata mirrors and mirrors provided in the builder spec for export/rebase

### CODE REVIEW

File: pkg/buildpod/generator.go [hunk 1]
```diff
@@ -56,54 +57,98 @@ const cnbUserId = "CNB_USER_ID"
...
-	err = json.Unmarshal([]byte(metadataJSON), &metadata)
+	var builderMetadata cnb.BuilderImageMetadata
+	err = json.Unmarshal([]byte(metadataJSON), &builderMetadata)
+	if err != nil {
+		return v1alpha1.BuildPodBuilderConfig{}, errors.Wrap(err, "unsupported builder metadata structure")
+	}
+
+	runImage, err := resolveRunImage(build, builderMetadata)
 	if err != nil {
 		return v1alpha1.BuildPodBuilderConfig{}, errors.Wrap(err, "unsupported builder metadata structure")
```
Comment (joaopapereira) [inline pkg/buildpod/generator.go]:
```suggestion
		return v1alpha1.BuildPodBuilderConfig{}, errors.Wrap(err, "unable to resolve Run Image")
```

File: pkg/cnb/cnb_metadata.go [hunk 2]
```diff
@@ -27,10 +27,15 @@ type BuilderImageMetadata struct {
 }
 
 type Stack struct {
-	RunImage string
+	RunImage RunImage
 	ID       string
 }
 
+type RunImage struct {
```
Comment (matthewmcnew) [inline pkg/cnb/cnb_metadata.go]: Can we annotate this with our desired json key?

**Comment by matthewmcnew** [comment]
Blocking merging this until https://github.com/buildpack/lifecycle/pull/202 is released to provide support for run images with a fully qualified digest.

**Comment by matthewmcnew** [comment]
This is significantly out-of-date. Closing.

**Pull request closed without merging**